classes/observation_manager.php

```diff
@@ -296,4 +298,96 @@ public static function reorder_observation_point(int $observationid, int $obpoin
 
         $transaction->allow_commit();
     }
+
+    /**
+     * Returns a list of observation points with responses for a particular session
+     * @param int $observationid ID of the observation instance
+     * @param int $sessionid ID of the observation sesssion
+     * @return array array of observation points with responses for the given session
+     */
+    public static function get_points_and_responses(int $observationid, int $sessionid) {
+        global $DB;
+
+        // Selects all points for this observation,
+        // and attaches responses for the current session (if one exists).
+
+        $sql = 'SELECT pts.id as point_id, obs_id, title, list_order, ins, ins_f, max_grade, res_type,
+        sess_resp.id as response_id, obs_ses_id as session_id, grade_given, response, ex_comment
+        FROM {observation_points} pts
+        LEFT JOIN
+        (SELECT * FROM {observation_point_responses} resp WHERE obs_ses_id = :sessionid) as sess_resp
```

Comment:
This subquery looks a little odd. you should be able to do something like

`LEFT JOIN {observation_point_responses} resp ON pts.id = sess_resp.obs_pt_id AND resp.obs_ses_id = :sessionid`

Multi condition joins are completely valid